MOM6/extra_sources/mom_cap_gtracer_flux.F90

```diff
@@ -0,0 +1,674 @@
+      !< Structure containing the gas fluxes between the atmosphere and the ocean and
+      !! parameters related to the calculation of these fluxes.
+  real, intent(in)                         :: ice_fraction(isc:iec,jsc:jec) !< sea ice fraction
+  integer, intent(in)                      :: isc !< The start i-index of cell centers within
+                                                  !! the computational domain
+  integer, intent(in)                      :: iec !< The end i-index of cell centers within the
+                                                  !! computational domain
+  integer, intent(in)                      :: jsc !< The start j-index of cell centers within
+                                                  !! the computational domain
+  integer, intent(in)                      :: jec !< The end j-index of cell centers within the
+                                                  !! computational domain
+  real, intent(in), optional               :: tsurf(isc:iec,jsc:jec) !< surface temperature
+  real, intent(in), optional               :: ustar(isc:iec,jsc:jec) !< friction velocity, not
+                                                                      !! used
+  real, intent(in), optional               :: cd_m (isc:iec,jsc:jec) !< drag coefficient, not
+                                                                      !! used
+
+  ! local variables
+  character(len=*), parameter   :: sub_name = 'atmos_ocean_fluxes_calc'
+  character(len=*), parameter   :: error_header =&
+      & '==>Error from ' // trim(mod_name) // '(' // trim(sub_name) // '):'
+  real, parameter                         :: permeg=1.0e-6
+
+  integer                                 :: n
+  integer                                 :: i
+  integer                                 :: j
+  real, dimension(:,:), allocatable       :: kw
+  real, dimension(:,:), allocatable       :: cair
+  character(len=128)                      :: error_string
+
+  ! Return if no fluxes to be calculated
+  if (gas_fluxes%num_bcs .le. 0) return
+
+  if (.not. associated(gas_fluxes%bc)) then
+    if (gas_fluxes%num_bcs .ne. 0) then
+      call mpp_error(FATAL, trim(error_header) // ' Number of gas fluxes not zero')
+    else
+      return
+    endif
+  endif
+
+  do n = 1, gas_fluxes%num_bcs
+    ! only do calculations if the flux has not been overridden
+    if ( .not. gas_fluxes%bc(n)%field(ind_flux)%override) then
+      if (gas_fluxes%bc(n)%flux_type .eq. 'air_sea_gas_flux_generic') then
+        if (.not. allocated(kw)) then
+          allocate( kw(isc:iec,jsc:jec) )
+          allocate ( cair(isc:iec,jsc:jec) )
+        elseif ((size(kw(:,:), dim=1) .ne. iec-isc+1) .or. (size(kw(:,:), dim=2) .ne. jec-jsc+1)) then
+          call mpp_error(FATAL, trim(error_header) // ' Sizes of flux fields do not match')
+        endif
+
+        if (gas_fluxes%bc(n)%implementation .eq. 'ocmip2') then
```

Comment:
I find this kind of multiple `if` statements easier to read if they are written as a `select case`:

```Fortran
select case (gas_fluxes%bc(n)%implementation)
case ('ocmip2')
...
case ('duce')
...
case ('johnson')
...
case default
...
end select
```

Reply:
I agree, but this is code that I copied from FMScoupler and I'd like to keep it as close as possible to the original (in case the original is updated)